packages/block-library/src/block/index.php

```diff
@@ -78,13 +78,48 @@ function render_block_core_block( $attributes ) {
 	 * filter so that it is available when a pattern's inner blocks are
 	 * rendering via do_blocks given it only receives the inner content.
 	 */
-	$has_pattern_overrides = isset( $attributes['content'] );
+	$has_pattern_overrides = isset( $attributes['content'] ) && null !== get_block_bindings_source( 'core/pattern-overrides' );
 	if ( $has_pattern_overrides ) {
 		$filter_block_context = static function ( $context ) use ( $attributes ) {
 			$context['pattern/overrides'] = $attributes['content'];
 			return $context;
 		};
 		add_filter( 'render_block_context', $filter_block_context, 1 );
+
+		// Add bindings for the default pattern overrides source.
+		$apply_default_pattern_overrides_bindings = static function ( $parsed_block ) {
+			$supported_block_attrs = array(
+				'core/paragraph' => array( 'content' ),
+				'core/heading'   => array( 'content' ),
+				'core/image'     => array( 'id', 'url', 'title', 'alt' ),
+				'core/button'    => array( 'url', 'text', 'linkTarget', 'rel' ),
+			);
```

Comment:
I understand that right now things are in flux and it's more convenient to hardcode or keep certain code paths private, but what's the plan for these mappings in the future?

* An attribute property in the schema? `{ "attributes:" { "content": { "...": "...", "overridable": true } } }`
* A block-supports record? `{ "supports": { "overrides": [ "content" ] } }`?
* Something entirely different?

Reply:
I think we've briefly discussed this before but decided to leave this for the future when we have a clear understanding of what the Block Binding API would look like. I believe the general consensus for the first version of pattern overrides is to use a hard-coded list.